libraries/Microsoft.Bot.Builder.AI.Orchestrator/OrchestratorRecognizer.cs

```diff
@@ -0,0 +1,346 @@
+            {
+                // add value
+                JToken values;
+                if (!recognizerResult.Entities.TryGetValue(entityResult.Type, StringComparison.OrdinalIgnoreCase, out values))
+                {
+                    values = new JArray();
+                    recognizerResult.Entities[entityResult.Type] = values;
+                }
+
+                // The Entity type names are not consistent, map everything to camelcase so we can process them cleaner.
+                var entity = JObject.FromObject(entityResult);
+                ((JArray)values).Add(entity.GetValue("text", StringComparison.InvariantCulture));
+
+                // get/create $instance
+                JToken instanceRoot;
+                if (!recognizerResult.Entities.TryGetValue("$instance", StringComparison.OrdinalIgnoreCase, out instanceRoot))
+                {
+                    instanceRoot = new JObject();
+                    recognizerResult.Entities["$instance"] = instanceRoot;
+                }
+
+                // add instanceData
+                JToken instanceData;
+                if (!((JObject)instanceRoot).TryGetValue(entityResult.Type, StringComparison.OrdinalIgnoreCase, out instanceData))
+                {
+                    instanceData = new JArray();
+                    instanceRoot[entityResult.Type] = instanceData;
+                }
+
+                JObject instance = new JObject();
+                instance.Add("startIndex", entity.GetValue("start", StringComparison.InvariantCulture));
+                instance.Add("endIndex", entity.GetValue("end", StringComparison.InvariantCulture));
+                instance.Add("score", (double)1.0);
+                instance.Add("text", entity.GetValue("text", StringComparison.InvariantCulture));
+                instance.Add("type", entity.GetValue("type", StringComparison.InvariantCulture));
+                instance.Add("resolution", entity.GetValue("resolution", StringComparison.InvariantCulture));
+                ((JArray)instanceData).Add(instance);
+            }
+
+            return recognizerResult;
+        }
+
+        private void InitializeModel()
+        {
+            if (_modelPath == null)
+            {
+                throw new ArgumentNullException($"Missing `ModelPath` information.");
+            }
+
+            if (_snapshotPath == null)
+            {
+                throw new ArgumentNullException($"Missing `ShapshotPath` information.");
+            }
+
+            if (orchestrator == null)
+            {
+                var fullModelPath = Path.GetFullPath(PathUtils.NormalizePath(_modelPath));
+                Stopwatch sw = new Stopwatch();
+
+                // Create Orchestrator 
+                try
+                {
+                    sw.Start();
+                    orchestrator = new Microsoft.Orchestrator.Orchestrator(fullModelPath);
+                }
+                catch (Exception ex)
```

Comment:
if you do want to leave the stopwatch in (in this case because this a once only heavy operation I think it's probably OK for now)

but at least do the "stop" in a "finally" on the try

Reply:
removed stop watches